gcloud_bigtable/row.py

```diff
@@ -416,12 +419,13 @@ def _commit_mutate(self, timeout_seconds=None):
             mutations=mutations_list,
         )
         timeout_seconds = timeout_seconds or self.timeout_seconds
-        response = self.client.data_stub.MutateRow.async(request_pb,
-                                                         timeout_seconds)
         # We expect a `._generated.empty_pb2.Empty`.
-        response.result()
+        if async:
+            self.client.data_stub.MutateRow.async(request_pb, timeout_seconds).result()
```

Comment:
By including `.result()` here, you're blocking evaluation of this expression from completing before the RPC itself has completed - so despite the caller's having passed `async=True` (or allowed `async`'s default value of `True` to be used) the evaluation of this entire function will not termination until the RPC has terminated. Is that what you intend?